crates/astria-sequencer/src/ibc/ics20_transfer.rs

```diff
@@ -753,343 +797,388 @@ mod test {
+        .expect_err("unknown asset during transfer to bridge account should fail");
     }
 
     #[tokio::test]
-    async fn execute_ics20_transfer_to_user_account_is_source_not_refund() {
+    async fn receive_funds_on_user_account_of_source_asset() {
         let storage = cnidarium::TempStorage::new().await.unwrap();
         let snapshot = storage.latest_snapshot();
         let mut state_tx = StateDelta::new(snapshot.clone());
 
         let recipient_address = astria_address(&[1; 20]);
         let amount = 100;
-        let base_denom = "nootasset".parse::<Denom>().unwrap();
         state_tx
-            .put_ibc_channel_balance(
-                &"dest_channel".to_string().parse().unwrap(),
-                &base_denom,
-                amount,
-            )
+            .put_ibc_channel_balance(&packet().chan_on_b, nria(), amount)
             .unwrap();
 
-        let packet = FungibleTokenPacketData {
-            denom: format!("source_port/source_channel/{base_denom}"),
+        let packet_data = FungibleTokenPacketData {
+            denom: source_asset().to_string(),
             sender: String::new(),
             amount: amount.to_string(),
             receiver: recipient_address.to_string(),
             memo: String::new(),
         };
-        let packet_bytes = serde_json::to_vec(&packet).unwrap();
 
-        execute_ics20_transfer(
+        receive_tokens(
             &mut state_tx,
-            &packet_bytes,
-            &"source_port".to_string().parse().unwrap(),
-            &"source_channel".to_string().parse().unwrap(),
-            &"dest_port".to_string().parse().unwrap(),
-            &"dest_channel".to_string().parse().unwrap(),
-            false,
+            &Packet {
+                data: serde_json::to_vec(&packet_data).unwrap(),
+                ..packet()
+            },
         )
         .await
-        .expect("valid ics20 transfer to user account; recipient, memo, and asset ID are valid");
+        .unwrap();
 
-        let balance = state_tx
-            .get_account_balance(recipient_address, &base_denom)
+        let user_balance = state_tx
+            .get_account_balance(recipient_address, &nria())
             .await
             .expect("ics20 transfer to user account should succeed");
-        assert_eq!(balance, amount);
-        let balance = state_tx
-            .get_ibc_channel_balance(&"dest_channel".to_string().parse().unwrap(), &base_denom)
+        assert_eq!(user_balance, amount);
+        let escrow_balance = state_tx
+            .get_ibc_channel_balance(&packet().chan_on_b, &nria())
             .await
             .expect("ics20 transfer to user account from escrow account should succeed");
-        assert_eq!(balance, 0);
+        assert_eq!(escrow_balance, 0);
     }
 
     #[tokio::test]
-    async fn execute_ics20_transfer_to_user_account_is_source_refund() {
+    async fn refund_user_account_with_source_zone_asset() {
```

Comment:
also add a test for `refund_user_account_with_sink_zone_asset`?

Reply:
Done in [5231ec7](https://github.com/astriaorg/astria/pull/1495/commits/5231ec7fb31d3a3a45b6cd56189aae9ad2465d99)